Command optimizing

itsDkiller opened the pull request:
### Edits I've done:

- [x] Removed personal references on file source headers
- [x] Changed to responses and error-notes of all commands to the vanilla equivalent
- [x] Optimized wierd handlers in some commands that where based on the "non-color-able" console
- [x] Removed the sender instance checker in the /stop-command file because the command will be useable by players
- [x] Removed the /chat command and its reference in command-manager.js as it is not a vanilla command
- [x] Added /version, /about and /list commands

itsDkiller commented:
The pull request is now completed, please let me know if there are issues or something else before merging

TheArmagan commented:
Please use [camelCase](https://en.wikipedia.org/wiki/Camel_case) for variable names.

TheArmagan commented:
Also add getRaknet() to Prismarine object for getting name data so you can get the max player amount for list command.
`sender.getServer().getRaknet().name.getMaxPlayerCount()`

TheArmagan commented:
```
getRaknet() {
   return this.#raknet
}
```
in Prismarine class

Bavfalcon9 commented on src/command/vanilla/version-command.js:
```diff
@@ -0,0 +1,28 @@
+     */
+    execute(sender, args) {
+        let server_version    = require('../../../package.json').version               //The only version reference I found, please contribute if there is a "better" one
+        let protocol_version  = require('../../network/identifiers').Protocol
```
Couldn't you just use one import here, rather than being spammy with `require().version`?

itsDkiller replied:
Yes, I could. I didn't rather that as too spammy.

TheArmagan commented on src/command/vanilla/kick-command.js:
```diff
@@ -12,21 +8,23 @@ class KickCommand extends Command {
-        if (args.length < 1 || args.length > 2) {
-            logger.warn('Invalid format, please use /kick <identifier> <reason>')
+        if (args.length < 1) {
+            logger.warn('§cYou have to specify a player.')
```
why here is `logger.warn` instead of `sender.sendMessage`?

TheArmagan commented on src/command/vanilla/tell-command.js:
```diff
@@ -20,25 +16,25 @@ class TellCommand extends Command {
         let message = args.splice(1).join(' ')
-        sender.sendMessage(`§7§oMe -> ${targetPlayer.name}: ${message}`)
-        targetPlayer.sendMessage(`§7§o${sender.name} -> Me: ${message}`)
+        sender.sendMessage(`§e[§f${sender.name} §e->§f ${targetPlayer.name}§e]§f ${message}`)
```
and you can make that like it:

**let messageToSend = `§e[§f${sender.name} §e->§f ${targetPlayer.name}§e]§f ${message}`**
**sender.sendMessage(messageToSend)**
**targetPlayer.sendMessage(messageToSend)**

for performance improvement

TheArmagan replied:
and you can make that like it:
```js
let messageToSend = `§e[§f${sender.name} §e->§f ${targetPlayer.name}§e]§f ${message}` 
sender.sendMessage(messageToSend)
targetPlayer.sendMessage(messageToSend)
```
for performance improvements..

itsDkiller commented:
- [x] Updated variable names to camelCase
- [x] Added getRaknet function to the prismarine instance
- [x] Fixed TODO in list-command.js (add max player count in the message)
- [x] Updated messages to send in tell-command.js and kick-command.js

TheArmagan commented:
Kick command crashes the server.
```js

D:\BDT\Downloads\JSPrismarine-command_optimizing\src\prismarine.js:184
            if (player.name.toLowerCase() === name.toLowerCase()) return player
                                                   ^

TypeError: Cannot read property 'toLowerCase' of undefined
    at Prismarine.getPlayerByName (D:\BDT\Downloads\JSPrismarine-command_optimizing\src\prismarine.js:184:52)
    at KickCommand.execute (D:\BDT\Downloads\JSPrismarine-command_optimizing\src\command\vanilla\kick-command.js:24:47)
    at CommandManager.dispatchCommand (D:\BDT\Downloads\JSPrismarine-command_optimizing\src\command\command-manager.js:98:32)
    at Function.handle (D:\BDT\Downloads\JSPrismarine-command_optimizing\src\network\handler\command-request-handler.js:15:48)
    at Listener.<anonymous> (D:\BDT\Downloads\JSPrismarine-command_optimizing\src\prismarine.js:79:33)
    at Listener.emit (events.js:314:20)
    at Connection.handlePacket (D:\BDT\Downloads\JSPrismarine-command_optimizing\node_modules\jsraknet\connection.js:477:28)
    at Connection.receivePacket (D:\BDT\Downloads\JSPrismarine-command_optimizing\node_modules\jsraknet\connection.js:302:22)
    at Connection.handleDatagram (D:\BDT\Downloads\JSPrismarine-command_optimizing\node_modules\jsraknet\connection.js:243:18)
    at Connection.receive (D:\BDT\Downloads\JSPrismarine-command_optimizing\node_modules\jsraknet\connection.js:179:25)
```

itsDkiller commented:
I'll take a look on that this evening, thank you

itsDkiller commented:
All commands have been tested again and are fully functional. Could you merge the PR if there are no other issues with this?

HerryYT commented:
Thank you so much! Merged 😄
